### PR TITLE
Pub 1312 Sensitivity Updates

### PR DESCRIPTION
**Pull request by ChrisS1512**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1312

### Change description ###

Updates to handle Sensitivity when returning requests. Data Management now makes a call to Account Management to determine whether the passed in user ID has permission to see the publication.

The verified flag on the endpoint has been replaced with this.

Also x-admin has now been added to the get flat file / payload endpoints, so that pub services is able to pull back to the file.

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java [hunk 1]
```diff
@@ -205,52 +205,79 @@ public Artefact getMetadataByArtefactId(UUID artefactId) {
...
-        } else {
-            throw new NotFoundException(String.format("No artefact found with the ID: %s", artefactId));
+        Optional<Artefact> artefact = artefactRepository.findByArtefactId(artefactId.toString(),
+            currentDate);
+
+        if (artefact.isPresent() && isAuthorised(artefact.get(), userId)) {
+            return artefact.get();
         }
+
+        throw new NotFoundException(String.format("No artefact found with the ID: %s", artefactId));
```
Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java:222]:
```suggestion
        throw new ArtefactNotFoundException(String.format("No artefact found with the ID: %s", artefactId));
```

Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java:222]:
```suggestion
        throw new ArtefactNotFoundException(String.format("No artefact found with the ID: %s", artefactId));
```

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/AccountManagementService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * Calls Account Management to determine whether a user is allowed to see a set publication.
+     * @param userId The UUID of the user to retrieve.
+     * @param listType The list type of the publication.
+     * @param sensitivity The sensitivity of the publication
+     * @return A flag indicating whether the user is authorised.
+     */
+    public Boolean getIsAuthorised(UUID userId, ListType listType, Sensitivity sensitivity) {
+        try {
+            return webClient.get().uri(new URI(String.format(
```
[the remark this answers is not in the document]
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/AccountManagementService.java]: Removed the use of URI

[the remark this answers is not in the document]
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/AccountManagementService.java]: Removed the use of URI

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/AccountManagementService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                "%s/account/isAuthorised/%s/%s/%s", url, userId, listType, sensitivity)))
+                .attributes(clientRegistrationId("accountManagementApi"))
+                .retrieve().bodyToMono(Boolean.class).block();
+        } catch (WebClientException | URISyntaxException ex) {
+            log.error(String.format("Request failed with error message: %s", ex.getMessage()));
```
[the remark this answers is not in the document]
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/AccountManagementService.java]: Now updated

[the remark this answers is not in the document]
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/AccountManagementService.java]: Now updated

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java (hunk 1 above)
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java]: Updated

**Review by joshblackmoor — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java [hunk 4]
```diff
@@ -312,10 +320,21 @@ public ResponseEntity<String> getArtefactPayload(
...
+        Resource file;
+        Artefact metadata;
+        if (isAdmin) {
+            file = publicationService.getFlatFileByArtefactID(artefactId);
+            metadata = publicationService.getMetadataByArtefactId(artefactId);
+        } else {
+            file = publicationService.getFlatFileByArtefactID(artefactId, userId);
+            metadata = publicationService.getMetadataByArtefactId(artefactId, userId);
+        }
+
```
[the remark this answers is not in the document]
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java:337]: Yes, you are right. Both options will work. That's fine with me :)

**Review by kurt-uk — APPROVED** [review]

**Review by junaidiqbalmoj — APPROVED** [review]
LGTM...

**Pull request merged**